### DISCUSSION TITLE
How to run executable scripts?

### DISCUSSION THREAD

**Discussion by flexid2000** [tsujan/FeatherPad#725] [Q&A] [answered]
Since long there has been the option 'run executable scripts'. However, I don't manage to get it working. Has anybody got an idea for the syntax rules for that string? How do you address the current file? With `%s`, `%b` or `%f`? Do you have to wrap it in a shell script, e.g. `sh -c "/path/to/script.sh %f"`?

Anyway, the help file says that both a button in the toolbar and in the file menu would appear. In my case, it doesn't. The Lubuntu help talks of the shortcut `Ctrl + E` to run the script. In my case, nothing happens. Any suggestions?

**Notes**

- This applies both to the Qt5 version 1.3.5-2 and the Qt6 version 1.3.5-1.
- The argument `%{Document:FilePath}` taken from the Kate editor doesn't work either.

**Comment by tsujan** [comment]
> ...nothing happens

As the text of the option says, it happens when the file you open is an *executable* script (written in a language like bash, python, ruby, lua or perl). If it isn't executable, nothing happens.

> ... the syntax rules for that string?

It's straightforward: any command that works on a file. But, because we deal with executable script files here, it's usually a command that executes the file inside a terminal — like `xterm -hold` for running in xterm.

Usually, you can leave it empty (the default), as explained in the tooltip.

> How do you address the current file?

You don't. The current file is always the executable script that has been opened in the current tab.

> Do you have to wrap it in a shell script

No, because the file is supposed to be executable.

**Reply by flexid2000** [reply]
Sorry for my misunderstanding. I had assumed the 'executable script' were an external program which could be run on the file opened in the editor. Kate for example does have this feature. That is quite handy for a certain workflow, e.g. when you write documents in markdown which are then automatically converted via pandoc to PDF. At least for me, that would be a nice feature for fpad, what do you think?

**Reply by tsujan** [reply]
> documents in markdown which are then automatically converted via pandoc to PDF

A Markdown document isn't an executable script. FeatherPad's "Run executable scripts" is exactly what its name says. What's executed is provided by the opened file, not by FeatherPad — the "command" is just auxiliary and can be omitted.

Supporting general commands is possible, and I might think about it if I find the time, but I prefer to stick with the Unix philosophy: "Write programs that do one thing..."

**Reply by flexid2000** [reply, marked as answer]
I get your point but still, I would appreciate it. It's just about connecting programs which do one thing ;)
